plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/ray_launcher.py

```diff
@@ -0,0 +1,211 @@
+        if self.mode != RayLauncherMode.LOCAL:
+            self.ray_cluster_config = cluster_config.get("ray_cluster_config")
+            self.ray_yaml_path = self._save_cluster_config()
+            self.docker_enabled = (
+                self.ray_cluster_config.get("docker").get("image") != ""
+            )
+
+    def setup(
+        self,
+        config: DictConfig,
+        config_loader: ConfigLoader,
+        task_function: TaskFunction,
+    ) -> None:
+        self.config = config
+        self.config_loader = config_loader
+        self.task_function = task_function
+
+    def launch(
+        self, job_overrides: Sequence[Sequence[str]], initial_job_idx: int
+    ) -> Sequence[JobReturn]:
+        setup_globals()
+        assert self.config is not None
+        assert self.config_loader is not None
+        assert self.task_function is not None
+
+        configure_log(self.config.hydra.hydra_logging, self.config.hydra.verbose)
+        sweep_dir = Path(str(self.config.hydra.sweep.dir))
+        sweep_dir.mkdir(parents=True, exist_ok=True)
+        log.info("Ray Launcher is launching {} jobs".format(len(job_overrides)))
+        log.info("Sweep output dir : {}".format(sweep_dir))
+
+        tmp_job_dump_dir = tempfile.mkdtemp()
+        for idx, overrides in enumerate(job_overrides):
+            idx = initial_job_idx + idx
+            log.info("\t#{} : {}".format(idx, " ".join(filter_overrides(overrides))))
+            sweep_config = self.config_loader.load_sweep_config(
+                self.config, list(overrides)
+            )
+            with open_dict(sweep_config):
+                # This typically coming from the underlying scheduler (SLURM_JOB_ID for instance)
+                # In that case, it will not be available here because we are still in the main process.
+                # but instead should be populated remotely before calling the task_function.
+                sweep_config.hydra.job.id = "job_id_for_{}".format(idx)
+                sweep_config.hydra.job.num = idx
+            HydraConfig.instance().set_config(sweep_config)
+            singleton_state = Singleton.get_state()
+            self._pickle_job(
+                idx,
+                tmp_job_dump_dir,
+                sweep_config=sweep_config,
+                config=self.config,
+                task_function=self.task_function,
+                singleton_state=singleton_state,
+            )
+
+        return self.launch_jobs(tmp_job_dump_dir)
+
+    def launch_jobs(self, tmp_job_dump_dir: str) -> Sequence[JobReturn]:
+        script_path = os.path.join(os.path.dirname(__file__), "ray_remote_invoke.py")
+        self._set_up_env()
+        tmp_local_job_return_dump_dir = tempfile.mkdtemp()
+        if self.mode == RayLauncherMode.LOCAL:
+            out, err = self._run_command(
+                ["python", script_path, tmp_job_dump_dir, tmp_local_job_return_dump_dir]
+            )
+            log.info(out)
```

Comment:
not loving this log here. it would only print everything at once at the end.
it's best to allow jobs to print to stdout as they run (with a pid). see the joblib plugin.